pywemo/ouimeaux_device/dimmer.py

```diff
@@ -1,37 +1,13 @@
-from . import Device
+from .switch import Switch
 
-
-class Dimmer(Device):
-
-    def set_state(self, state):
-        """
-        Set the state of this device to on or off.
-        """
-        self.basicevent.SetBinaryState(BinaryState=int(state))
-        self._state = int(state)
-
-    def off(self):
-        """
-        Turn this device off. If already off, will return "Error".
-        """
-        return self.set_state(0)
-
-    def on(self):
-        """
-        Turn this device on. If already on, will return "Error".
-        """
-        return self.set_state(1)
-
-    def toggle(self):
-        """
-        Toggle the dimmer's state.
-        """
-        return self.set_state(not self.get_state())
-		
+class Dimmer(Switch):		
     def set_brightness(self, brightness):
         """
         Set the brightness of this device to an integer between 1-100.
         """
+        if self.get_state() == 0 or None:
```

Comment:
Can I check what you want to do here?

Have you found that setting brightness on the device doesn't turn it on?

Reply:
setting brightness does not turn it on, this was some trial and error to make sure setting brightness turns device on as well. Sorry, somewhat new to GitHub, trying to get better at comments ;)

Reply:
In that case you may want to do something like:-
```

if brightness == 0:
    if self.get_state() != 0:
        self.off()
else
    if self.get_state() == 0:
        self.on()
```

Assuming you want to do an explicit turn off too.